### PR TITLE
TSPS-191 add b2c authentication

### PR DESCRIPTION
**Pull request by mmorgantaylor**
### Description 

Add b2c authentication to our swagger

Associated PRs: 
- register the dev b2c redirect url: https://github.com/broadinstitute/terraform-ap-deployments/pull/1672
- add environment variables: https://github.com/broadinstitute/terra-helmfile/pull/5806

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-191

### CODE REVIEW

File: service/src/test/java/bio/terra/pipelines/controller/PublicApiControllerTest.java [hunk 1]
```diff
@@ -60,4 +72,31 @@ void testVersion() throws Exception {
...
+    var swaggerPaths = Set.of("/index.html", "/swagger-ui.html");
+    for (var path : swaggerPaths) {
+      this.mockMvc
+          .perform(get(path))
+          .andExpect(status().isOk())
+          .andExpect(model().attributeExists("clientId"));
+    }
+  }
+
+  // this test doesn't work because I can't figure out how to use the secondaryTemplateResolver bean
```
Comment (mmorgantaylor) [inline service/src/test/java/bio/terra/pipelines/controller/PublicApiControllerTest.java:92]: happy to just take this out unless anyone knows how to make it work

Comment (jsotobroad) [reply service/src/test/java/bio/terra/pipelines/controller/PublicApiControllerTest.java:92]: what endpoint do you expect to use the secondaryTempalteResolver?

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -20,6 +20,10 @@ env:
     leonardo: ${LEONARDO_ADDRESS:https://leonardo.dsde-dev.broadinstitute.org}
     workspacemanager: ${WORKSPACE_MANAGER_ADDRESS:https://workspace.dsde-dev.broadinstitute.org}
     rawls: ${RAWLS_ADDRESS:https://rawls.dsde-dev.broadinstitute.org}
+  oidc:
+    clientId: ${OIDC_CLIENT_ID:bbd07d43-01cb-4b69-8fd0-5746d9a5c9fe}
```
Comment (mmorgantaylor) [inline service/src/main/resources/application.yml:24]: TODO: remove the default values and populate them in local env vars using a private script

File: service/src/main/java/bio/terra/pipelines/app/configuration/internal/TemplateResolvers.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package bio.terra.pipelines.app.configuration.internal;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+import org.thymeleaf.templatemode.TemplateMode;
+import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver;
+
+@Component
+public class TemplateResolvers {
```
Comment (mmorgantaylor) [inline service/src/main/java/bio/terra/pipelines/app/configuration/internal/TemplateResolvers.java:9]: move this to an `app/components/` directory

**Review by jsotobroad — APPROVED** [review]
:woohoo:

**Pull request merged**